src/main/java/org/mule/extension/sftp/internal/SftpInputStream.java

```diff
@@ -50,14 +53,22 @@ protected static ConnectionManager getConnectionManager(SftpConnector config) th
    * @throws ConnectionException if a connection could not be established
    */
   public static SftpInputStream newInstance(SftpConnector config, SftpFileAttributes attributes, PathLock lock,
-                                            Long timeBetweenSizeCheck)
+                                            Long timeBetweenSizeCheck, TimeUnit timeBetweenSizeCheckUnit)
       throws ConnectionException {
-    ConnectionManager connectionManager = getConnectionManager(config);
     ConnectionAwareSupplier connectionAwareSupplier =
-        new ConnectionAwareSupplier(attributes, connectionManager, timeBetweenSizeCheck, config);
+        new ConnectionAwareSupplier(attributes, getConnectionManager(config),
+                                    calculateTimeBetweenSizeCheckInMillis(timeBetweenSizeCheck, timeBetweenSizeCheckUnit),
+                                    config);
     return new SftpInputStream(connectionAwareSupplier, lock);
   }
 
+  private static Long calculateTimeBetweenSizeCheckInMillis(Long timeBetweenSizeCheck, TimeUnit timeBetweenSizeCheckUnit) {
+    if (timeBetweenSizeCheck == null || timeBetweenSizeCheck < 1) {
```

Comment:
`timeBetweenSizeCheck < 1` should not be mapped to "no timeBetweenSizeCheck". If it is not a valid value, fail with a clear message